util/config/make_sys_basic_types.py

```diff
@@ -0,0 +1,230 @@
+        logging.debug('Creating temp header: {0}'.format(h_file))
+        with open(h_file, 'w') as fp:
+            fp.write("""
+#include "sys_basic.h"
+#include <limits.h>
+#include <stdint.h>
+#include <math.h>
+
+FIND_INT_SIZES_START
+""")
+
+            for max_macro, _, _ in _types:
+                fp.write('{0}\n'.format(max_macro))
+            logging.debug('Wrote {0} max types to {1}'.format(len(_types), h_file))
+
+        compile_cmd = '{compileline} -E {h_file}'.format(**locals())
+        logging.debug('Preprocessor command: {0}'.format(compile_cmd))
+        compile_result = os.popen(compile_cmd).read()
+        logging.debug('Captured preprocessor output with {0} '
+                      'characters.'.format(len(compile_result)))
+
+    # Iterate through the preprocessor output, find the max value expressions,
+    # and record them in a list.
+    max_exprs = []
+    keep = False
+    for line in compile_result.splitlines():
+        if line.startswith('#') or line.startswith('//'):
+            continue
+        elif 'FIND_INT_SIZES_START' in line:
+            keep = True
+        elif keep:
+            max_exprs.append(line.strip())
+    logging.debug('Found {0} lines of max type '
+                  'values.'.format(len(max_exprs)))
+
+    if len(max_exprs) != len(_types):
+        logging.error('Found {0} max values, but {1} types were '
+                      'expected.'.format(len(max_exprs), len(_types)))
+        sys.exit(1)
+
+    # Iterate through the max value expressions, evaluate each one, and store
+    # it in a list. Python deals with arbitrarily large integers, so there is
+    # no fear of overflow.
+    replace_pattern = re.compile(r'[UL]', re.IGNORECASE)
+    max_values = []
+    for expr in max_exprs:
+        ex = re.sub(replace_pattern, '', expr)
+        value = eval(ex)
+        logging.debug('{0} -> {1}'.format(expr, value))
+        max_values.append(value)
+    logging.debug('Evaluated all {0} expressions from '
+                  'preprocessor.'.format(len(max_values)))
+
+    # Iterate through the chapel types/max values and print out the SysCTypes
+    # Chapel module code. Each line takes the form "extern type <chpl_type>=
+    # <chpl_value>;" where <chpl_value> is found by looking up the max value
+    # (from evaluated expression above) in the _max_value_to_chpl_type map.
+    sys_c_types = []
+    handled_c_ptr = False
+    for i, max_value in enumerate(max_values):
+        max_macro, chpl_type, c_type = _types[i]
+        chpl_value = _max_value_to_chpl_type.get(str(max_value))
+        if chpl_value is None:
+            logging.error('Unknown numeric limit {0} in '
+                          '_max_value_to_chpl_type dict.'.format(max_value))
+            sys.exit(1)
+
+        sys_c_types.append('/* The type corresponding to the C {c_type} type'
+                           ' */'.format(**locals()))
+        stmt = 'extern type {chpl_type}= '.format(**locals())
```

Comment:
I remain uncertain of what .format(**locals()) actually does, but that's really my problem. Anyway...